docs/aws/audit/kubernetesmonitoring/rules/is_k8s_logging_enabled.mdx

```diff
@@ -24,6 +24,212 @@ HITRUST, SOC2, NISTCSF, PCIDSS
+       name='my-eks-cluster',  # Replace with your desired cluster name
+       version='1.21',  # Replace with your desired Kubernetes version
+       roleArn='arn:aws:iam::123456789012:role/EKSClusterRole',  # Replace with your IAM role ARN
+       resourcesVpcConfig={
+           'subnetIds': [
+               'subnet-abcde123',  # Replace with your subnet IDs
+               'subnet-abcde456'
+           ],
+           'securityGroupIds': [
+               'sg-abcde123'  # Replace with your security group IDs
+           ]
+       },
+       logging=logging_config
+   )
+
+   print("Cluster creation initiated with logging enabled:", response)
+   ```
+
+4. **Run the Script**:
+   Execute the script to create the EKS cluster with logging enabled.
+   ```bash
+   python create_eks_cluster_with_logging.py
+   ```
+
+This script ensures that the EKS cluster is created with logging enabled for API, audit, authenticator, controller manager, and scheduler logs. Adjust the parameters such as region, cluster name, Kubernetes version, IAM role ARN, subnet IDs, and security group IDs as per your requirements.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Amazon EKS console at https://console.aws.amazon.com/eks/home#/clusters.
+3. Select the EKS cluster you want to check.
+4. In the cluster details page, under the "Logging" section, check if the "CloudWatch Logs" are enabled. If the status is "Enabled", it means that logging is enabled for the EKS cluster. If the status is "Disabled", it means that logging is not enabled for the EKS cluster.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: To interact with AWS services, you need to install the boto3 library. You can install it using pip:
+
+   ```bash
+   pip install boto3
+   ```
+
+2. Configure AWS credentials: Before you can interact with AWS services, you need to set up your AWS credentials. You can do this by setting the following environment variables:
+
+   ```bash
+   export AWS_ACCESS_KEY_ID='your_access_key'
+   export AWS_SECRET_ACCESS_KEY='your_secret_key'
+   ```
+
+3. Create a Python script to check the logging configuration of EKS clusters:
+
+   ```python
+   import boto3
+
+   def check_eks_logging():
+       client = boto3.client('eks')
+       clusters = client.list_clusters()['clusters']
+       for cluster in clusters:
+           response = client.describe_cluster(name=cluster)
+           logging = response['cluster']['logging']['clusterLogging']
+           for log in logging:
+               if log['enabled']:
+                   print(f"Logging is enabled for {log['types']} in {cluster}")
+               else:
+                   print(f"Logging is not enabled for {log['types']} in {cluster}")
+
+   if __name__ == '__main__':
+       check_eks_logging()
+   ```
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>